service/src/main/kotlin/com/sourceclear/agile/piplanning/service/controllers/BoardControllerK.kt

```diff
@@ -272,48 +272,54 @@ open class BoardControllerK @Autowired constructor(
 
     val s = create.fetchOne(STORY_REQUESTS, STORY_REQUESTS.ID.eq(requestId))
         ?: throw notFound
-    if (boardId != s.toBoardId) { // has to be sent from the receiving board
-      throw unauthorized
-    }
 
-    if (valueOf(s.state) == Pending) {
-      s.state = Accepted.name
+    if (s.state != Pending.name) {
+      // for acknowledgements
+      create.update(NOTIFICATIONS)
+          .set(NOTIFICATIONS.ACKNOWLEDGED, true)
+          .where(NOTIFICATIONS.STORY_REQUEST_ID.eq(requestId).and(NOTIFICATIONS.TYPE.notEqual(NotificationO.IncomingStoryRequest::class.simpleName)))
+          .execute()
```

Comment:
Maybe I'm misunderstanding this, but if this is triggered on anything but a pending request, it's probably an FE error and we should fail immediately (the `badRequest` we had previously). Why do we acknowledge in that case?

Reply:
`storyRequestId` is required here https://github.com/srcclr/sapling-frontend/blob/master/src/components/NotificationsList.tsx#L19

I am reusing the `/accept` path for user's acknowledgement for non-pending story requests.
e.g withdraw -> acknowledge by click on acknowledgement button on frontend
e.g accepted -> acknowledge by click on acknowledgement button on frontend
e.g rejected   -> acknowledge by click on acknowledgement button on frontend

Reply:
Could the existing /acknowledge endpoint be used for this?

Reply:
🤦 , thanks for pointing that out.